File: enterprise/approvals/models/mail_activity.py
# -*- coding: utf-8 -*-
# Part of Odoo. See LICENSE file for full copyright and licensing details.

from odoo import models
from odoo.addons.mail.tools.discuss import Store


class MailActivity(models.Model):
    _inherit = "mail.activity"

    def _to_store(self, store: Store):
        super()._to_store(store)
        activity_type_approval_id = self.env.ref("approvals.mail_activity_data_approval")
        for activity in self.filtered(
            lambda activity: activity["res_model"] == "approval.request"
            and activity.activity_type_id == activity_type_approval_id
        ):
            request = self.env["approval.request"].browse(activity["res_id"])
            approver = request.approver_ids.filtered(
                lambda approver: activity.user_id == approver.user_id
            )
            store.add(activity, {"approver_id": approver.id, "approver_status": approver.status})

File: enterprise/approvals/static/tests/mock_server/mock_models/mail_activity.js
import { mailModels } from "@mail/../tests/mail_test_helpers";

export class MailActivity extends mailModels.MailActivity {
    /** @param {number[]} ids */
    _to_store(ids, store) {
        super._to_store(...arguments);
        for (const activity of this._filter([
            ["id", "in", ids],
            ["res_model", "=", "approval.request"],
        ])) {
            // check on activity type being approval not done here for simplicity
            const [approver] = this.env["approval.approver"]._filter([
                ["request_id", "=", activity.res_id],
                ["user_id", "=", activity.user_id],
            ]);
            if (approver) {
                store.add(this.browse(activity.id), {
                    approver_id: approver.id,
                    approver_status: approver.status,
                });
            }
        }
    }
}
